### ISSUE TITLE
How do you use exact phase matching in keyword searches?

### ISSUE THREAD

**Issue by brandonkopp** [fedspendingtransparency/usaspending-api#2918]
I am attempting to use the `/api/v2/search/spending_by_award/` endpoint and I would like to use exact phrase matching rather than AND or OR matching for keyword search. For example, I would like to search for "data science" and return matches containing only that exact phrase but the current search functionality, as I understand it, returns entries that have either "data" or "science" which includes many, many matches that are outside of the scope of what I am interested in.

I have attempted to escape an inner set of double quotes and experimented with some elastisearch functionality (e.g., match_phrase) but it looks like you sanitize the inputs and I end up getting a failure code.

Thank you.

**Comment by emily-brents** [comment]
Hi Brandon,

If you are looking to match an exact phrase, you can use `+` to connect the words in the phrase you are trying to match. For example, if you were looking for "data science", you should enter `data+science` as your search text.

**Comment by brandonkopp** [comment]
Thank you for your response. 

Using `+` seems to undershoot what I'm looking for. When I use that operator on the `search_by_award` endpoint with keyword `data+science`, I get 1 result. The `search_by_award_count` endpoint with keyword `data+science` shows the same thing.

Perhaps there's something I'm doing wrong? Below is a minimal example that returns 1 record when I have found several records including "data science" using the search functions on usaspending.gov. Using `data science` returns more than 11,000 records, many of which that don't appear to have "data science" as a phrase. 

```{python}
import requests

def get_usaspending_api_return(url, body):
    r = requests.post(url = url, json = body) 
    
    if r.status_code == 200:
        return(r.json())
    else:
        raise Exception(f"Failure - Status Code: {r.status_code}")
        
body = {"subawards": "false",
          "filters": {
              "keyword": "data+science",
              "award_type_codes": ["A", "B", "C","D"]
          },
          "fields": [
              "Award ID","Recipient Name","Recipient DUNS Number","Start Date","End Date","Last Modified Date",
              "Award Amount","Awarding Agency","Awarding Sub Agency","Contract Award Type",
              "Funding Agency","Funding Sub Agency","Description",
              "Place of Performance State Code","Place of Performance Zip5"
          ],
          "page": 1,
          "limit": 10,
          "sort": "Last Modified Date",
          "order": "desc"
      }

url= "https://api.usaspending.gov/api/v2/search/spending_by_award/"
data = get_usaspending_api_return(url, body)
```

I don't want to get overly hung up on the example "data science". It is meant here as an illustrative example because "data" and "science" are both common words so are likely to return a lot of non-target matches when searching for an exact phase.

**Comment by emily-brents** [comment]
It does seem like using the `+` operator does seem to restrict the results more than would be necessary. This could be due to a bug, which we will investigate. In the meantime, I would recommend using the keyword search feature, which allows for more precise searches on phrases. the endpoint for that search is `/api/v2/search/spending_by_transaction/`. Please note that as the name indicates, this search uses transaction data, rather than the award data, but the award ID will be linked in the results if you are looking for the data on the specific award, rather than the transaction.
Here is an example request for the `spending_by_transaction` endpoint - for exact matches you should place the string you are looking to match in quotes `"`, which will need to be escaped.
```
{
    "filters": {
        "keyword": "\"data science\"",
        "award_type_codes": [
            "A",
            "B",
            "C",
            "D"
        ]
    },
    "fields": [
        "Award ID",
        "Mod",
        "Recipient Name",
        "Action Date",
        "Transaction Amount",
        "Awarding Agency",
        "Awarding Sub Agency",
        "Award Type"
    ],
    "page": 1,
    "limit": 35,
    "sort": "Transaction Amount",
    "order": "desc"
}
```
Thank you for bringing this to our attention.

**Comment by brandonkopp** [comment]
Thank you Emily.  The thing I'm most interested in is the "Description" field and that doesn't appear to be accessible with the `spending_by_transaction` endpoint. I made a less than ideal workaround using the `spending_by_award` endpoint which takes in all of the data and then uses Python to filter the dataset. For 10 years worth of data on "data science" that results in 96 calls to the API to collect all 9600+ matches for a net of about 150 final, exact matches, but it works. I only need to do that once to get historical data and then for ongoing collection, I will just get the most recent 100 matches in a single call and filter from there.

I appreciate your work in putting this API together and your thoughtful responses to my issue. This is the first API I've used that uses the post format so it has been a great learning experience.
